led.lua

```diff
@@ -18,6 +18,20 @@ function M.animate(frames, s)
 end
 
 
+function M.run_around(frame, s)
+    s = s or M.speed
+    local co = coroutine.create(function()
+        while true do
+	   for i = 1, #frame, 3 do
+	      ws2812.write(string.sub(frame, i) .. string.sub(frame, 1, i))
```

Comment:
also lookup at ws2812 library for shifting, maybe it is more performant than doing this :)